### PR TITLE
MM-48214: basic makefile

### PR DESCRIPTION
**Pull request by ifoukarakis**
#### Summary

- [x] Update `Makefile` to match practices suggested by release team. 
- [x] Add python actions and help command to `Makefile`.
- [x] Removed not needed `Dockerfile`.
- [x] Moved `docker-compose.yml` used for spinning local environments under `./build/`.

Docker related options will be added as a separate PR, together with a proper `Dockerfile`.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,15 +1,144 @@
...
+data-image: ## to start a bash shell on the data image
+	$(AT)$(INFO) Attaching to data-image and mounting repo...
+	$(AT)export DBT_PROFILE_PATH=".";\
+	$(DOCKER) compose -f ${DOCKER_COMPOSE_DBT_FILE} run data_image bash
+
+.PHONY: clean
+clean: ## to clean-up
+	@$(INFO) cleaning ${PYTHON_OUT_BIN_DIR} folder...
+	$(AT)rm -rf ${PYTHON_OUT_BIN_DIR} || ${FAIL}
+	@$(OK) cleaning ${PYTHON_OUT_BIN_DIR} folder
```
Comment (phoinixgrr) [inline Makefile:144]: Let's add an NL here

File: Makefile [hunk 2]
```diff
@@ -1,15 +1,144 @@
...
+python-test: ## to run python tests
+	$(AT)$(INFO) testing python...
+	$(AT)$(POETRY) run pytest || ${FAIL}
+	$(AT)$(OK) testing python
+
+.PHONY: python-update-dependencies
+python-update-dependencies: ## to update python dependencies
+	$(AT)$(INFO) updating python dependencies...
+	$(AT)$(POETRY) install && \
+	    $(POETRY) run pip install clearbit==0.1.7 || ${FAIL}
```
Comment (phoinixgrr) [inline Makefile:113]: Would refrain from statically defining the version inside the target.
Maybe better defining this python package and it's version as a variable? What do you say?

File: Makefile [hunk 3]
```diff
@@ -1,15 +1,144 @@
...
+# Docker compose specific environment configuration file
+DOCKER_COMPOSE_ENV_FILE      := .dbt.env
+
+## Github Variables
+# A github access token that provides access to upload artifacts under releases
+GITHUB_TOKEN                 ?= a_token
+# Github organization
+GITHUB_ORG                   := mattermost
+# Most probably the name of the repo
+GITHUB_REPO                  := ${APP_NAME}
```
Comment (phoinixgrr) [inline Makefile:55]: Most of the variables here are unused. 
Let's clear them up to improve readability.

File: Makefile [hunk 4]
```diff
@@ -1,15 +1,144 @@
...
+	$(AT)$(POETRY) install && \
+	    $(POETRY) run pip install clearbit==0.1.7 || ${FAIL}
+	$(AT)$(OK) updating python dependencies
+
+.PHONY: dbt-docs
+dbt-docs: ## to generate and serve dbt docs
+	$(AT)$(INFO) Generating docs and spinning up the a webserver on port 8081...
+	$(AT)export DBT_PROFILE_PATH=".";\
+	$(DOCKER) compose run -p "8081:8081" dbt_image bash -c "dbt deps && dbt docs generate -t prod && dbt docs serve --port 8081"
+
```
Comment (phoinixgrr) [inline Makefile:121]: Would expect an `$(AT)$(OK)  Generating docs and spinning up the a webserver on port 8081...` to provide proper logging

File: Makefile [hunk 5]
```diff
@@ -1,15 +1,144 @@
...
+	$(AT)$(INFO) updating python dependencies...
+	$(AT)$(POETRY) install && \
+	    $(POETRY) run pip install clearbit==0.1.7 || ${FAIL}
+	$(AT)$(OK) updating python dependencies
+
+.PHONY: dbt-docs
+dbt-docs: ## to generate and serve dbt docs
+	$(AT)$(INFO) Generating docs and spinning up the a webserver on port 8081...
+	$(AT)export DBT_PROFILE_PATH=".";\
+	$(DOCKER) compose run -p "8081:8081" dbt_image bash -c "dbt deps && dbt docs generate -t prod && dbt docs serve --port 8081"
```
Comment (phoinixgrr) [inline Makefile:120]: Would expect a `|| ${FAIL}` here

File: Makefile [hunk 6]
```diff
@@ -1,15 +1,144 @@
...
+
+.PHONY: dbt-generate-docs
+dbt-generate-docs: ## to generate dbt docs
+	$(AT)$(INFO) Generating docs
+	$(AT)export DBT_PROFILE_PATH=".";\
+	$(DOCKER) compose run dbt_image bash -c "dbt deps && dbt docs generate -t prod"
```
Comment (phoinixgrr) [inline Makefile:126]: Would expect a `|| ${FAIL}` here

File: Makefile [hunk 7]
```diff
@@ -1,15 +1,144 @@
...
+
+.PHONY: dbt-bash
+dbt-bash: ## to start a bash shell with DBT
+	$(AT)$(INFO) Running bash with dbt...
+	$(AT)export DBT_PROFILE_PATH=".";\
+	$(DOCKER) compose -f ${DOCKER_COMPOSE_DBT_FILE} run dbt_image bash -c "dbt deps && /bin/bash"
```
Comment (phoinixgrr) [inline Makefile:132]: Would expect a `|| ${FAIL}` here

File: Makefile [hunk 8]
```diff
@@ -1,15 +1,144 @@
...
+
+.PHONY: data-image
+data-image: ## to start a bash shell on the data image
+	$(AT)$(INFO) Attaching to data-image and mounting repo...
+	$(AT)export DBT_PROFILE_PATH=".";\
+	$(DOCKER) compose -f ${DOCKER_COMPOSE_DBT_FILE} run data_image bash
```
Comment (phoinixgrr) [inline Makefile:138]: Would expect a `|| ${FAIL}` here

File: Makefile [hunk 9]
```diff
@@ -1,15 +1,144 @@
...
+	$(AT)$(INFO) Generating docs and spinning up the a webserver on port 8081...
+	$(AT)export DBT_PROFILE_PATH=".";\
+	$(DOCKER) compose run -p "8081:8081" dbt_image bash -c "dbt deps && dbt docs generate -t prod && dbt docs serve --port 8081"
+
+.PHONY: dbt-generate-docs
+dbt-generate-docs: ## to generate dbt docs
+	$(AT)$(INFO) Generating docs
+	$(AT)export DBT_PROFILE_PATH=".";\
+	$(DOCKER) compose run dbt_image bash -c "dbt deps && dbt docs generate -t prod"
+
```
Comment (phoinixgrr) [inline Makefile:127]: Would expect an $(AT)$(OK) ... to provide proper logging

File: Makefile [hunk 10]
```diff
@@ -1,15 +1,144 @@
...
+.PHONY: dbt-bash
+dbt-bash: ## to start a bash shell with DBT
+	$(AT)$(INFO) Running bash with dbt...
+	$(AT)export DBT_PROFILE_PATH=".";\
+	$(DOCKER) compose -f ${DOCKER_COMPOSE_DBT_FILE} run dbt_image bash -c "dbt deps && /bin/bash"
+
```
Comment (phoinixgrr) [inline Makefile:133]: Would expect an $(AT)$(OK) ... to provide proper logging

File: Makefile [hunk 11]
```diff
@@ -1,15 +1,144 @@
...
+.PHONY: data-image
+data-image: ## to start a bash shell on the data image
+	$(AT)$(INFO) Attaching to data-image and mounting repo...
+	$(AT)export DBT_PROFILE_PATH=".";\
+	$(DOCKER) compose -f ${DOCKER_COMPOSE_DBT_FILE} run data_image bash
+
```
Comment (phoinixgrr) [inline Makefile:139]: Would expect an $(AT)$(OK) ... to provide proper logging

**Review by phoinixgrr — COMMENTED** [review]
Awesome work! 🎖️ 

Happy to see that the template makefile that we developed, is getting used and helps teams :) 

I left some minor comments as improvements. 
Good idea to also have seperation in commits as described in the description of the PR
```
- Update Makefile to match practices suggested by release team.
- Add python actions and help command to Makefile.
- Removed not needed Dockerfile.
- Moved docker-compose.yml used for spinning local environments under ./build/.
```
That would help provide quicker and easier reviews, as Review by commit would be possible!

File: Makefile (hunk 2 above)
Comment (ifoukarakis) [reply Makefile:113]: I agree but unfortunately this is a really old dependency and it's the only way to install it. The plan is to remove the dependency as soon as we start improving the code.

File: Makefile (hunk 4 above)
Comment (ifoukarakis) [reply Makefile:121]: The original intent was to leave it as it was (docker compose starting containers and waiting for `ctrl+c` to finish). Made a change to start the container in detached mode. Also added a (slightly) more helpful OK log message.

**Comment by ifoukarakis** [comment]
> Awesome work! 🎖️
> 
> Happy to see that the template makefile that we developed, is getting used and helps teams :)
> 
> I left some minor comments as improvements. Good idea to also have seperation in commits as described in the description of the PR
> 
> ```
> - Update Makefile to match practices suggested by release team.
> - Add python actions and help command to Makefile.
> - Removed not needed Dockerfile.
> - Moved docker-compose.yml used for spinning local environments under ./build/.
> ```
> 
> That would help provide quicker and easier reviews, as Review by commit would be possible!

File: Makefile (hunk 2 above)
Comment (phoinixgrr) [reply Makefile:105]: Works for me as it is a temporary solution.

File: Makefile [hunk 12]
```diff
@@ -1,15 +1,140 @@
...
+	$(AT)$(INFO) Attaching to data-image and mounting repo...
+	$(AT)export DBT_PROFILE_PATH=".";\
+	$(DOCKER) compose -f ${DOCKER_COMPOSE_DBT_FILE} run data_image bash || ${FAIL}
+	$(AT)$(OK) Exited data-image
+
+.PHONY: clean
+clean: ## to clean-up
+	@$(INFO) cleaning ${PYTHON_OUT_BIN_DIR} folder...
+	$(AT)rm -rf ${PYTHON_OUT_BIN_DIR} || ${FAIL}
+	@$(OK) cleaning ${PYTHON_OUT_BIN_DIR} folder
```
Comment (phoinixgrr) [inline Makefile:140]: Any reason for not including the verbosity flag here ? `$(AT)` ?

Comment (ifoukarakis) [reply Makefile]: Most likely copy-paste error, fixed :)

**Review by phoinixgrr — APPROVED** [review]

File: Makefile (hunk 12 above)
Comment (phoinixgrr) [reply Makefile]: @ifoukarakis Thanks! 

Also there are some tests failing

**Pull request merged**